src/App.tsx

```diff
@@ -37,16 +40,21 @@ const App = () => {
   }, [isSubmitted, searchQuery]);
 
   return (
-    <>
-      <Header />
-      <Search
-        searchQuery={searchQuery}
-        setSearchQuery={setSearchQuery}
-        setIsSubmitted={setIsSubmitted}
-      />
-      <Main posts={posts} active={active} setActive={setActive} sort={sort} setSort={setSort} />
-      <Footer />
-    </>
+    <AppProvider>
+      <>
+        <Header />
+        <Search
+          searchQuery={searchQuery}
+          setSearchQuery={setSearchQuery}
+          setIsSubmitted={setIsSubmitted}
+        />
+        <Main posts={posts} active={active} setActive={setActive} sort={sort} setSort={setSort} />
+        <Footer />
+        <Modal>
```

Comment:
When you move Context provider outside content you will able to use conditional rendering for modal